### PR TITLE
Give DefaultAssetBundle a default asset bundle.

### PR DESCRIPTION
**Pull request by Hixie**
This makes it easier to use e.g. AssetImage in simple test applications.

Also, dartdoc improvements.

Also, use @required in one place, to see if it causes any trouble. If it
doesn't, I'll start using it in more places.

### CODE REVIEW

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/3341.

@HansMuller

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -2134,10 +2134,19 @@ class DefaultAssetBundle extends InheritedWidget {
...
+  /// at the given context, then this will return the [rootBundle].
+  /// The [rootBundle] does not automatically select images based on
+  /// the current device pixel ratio. To get an asset bundle that
+  /// automatically performs pixel-density-aware asset resolution, use
+  /// a [MaterialApp], [WidgetsApp], or [AssetVendor] widget, which
+  /// introduce a suitably-configured [DefaultAssetBundle] widget.
   static AssetBundle of(BuildContext context) {
     DefaultAssetBundle result = context.inheritFromWidgetOfExactType(DefaultAssetBundle);
-    return result?.bundle;
+    return result?.bundle ?? rootBundle;
```
Comment (eseidelGoogle) [inline packages/flutter/lib/src/widgets/basic.dart]: I suspect we'll end up wanting to namspace our globals at some point. :)

**Comment by eseidelGoogle** [comment]
lgtm

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: namespacing is under the control of the importer (import ... as).

Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: (e.g. we regularly namespaces dart:ui imports "as ui", and mojom imports "as mojom".)

**Pull request merged**